internal/operator-controller/authorization/rbac.go

```diff
@@ -58,59 +63,61 @@ func NewRBACPreAuthorizer(cl client.Client) PreAuthorizer {
-//
-// Return Value:
-//   - nil: indicates that the authorization check passed and the operation is permitted.
-//   - non-nil error: indicates that the authorization failed (either due to insufficient permissions
-//     or an error encountered during the check), the error provides a slice of several failures at once.
+// as defined by the RBAC policy. It decodes the manifest, constructs authorization records,
+// performs attribute checks, and then uses escalationChecker (which delegates to upstream logic)
+// to verify that no privilege escalation is occurring.
 func (a *rbacPreAuthorizer) PreAuthorize(ctx context.Context, manifestManager user.Info, manifestReader io.Reader) ([]ScopedPolicyRules, error) {
-	var allMissingPolicyRules = []ScopedPolicyRules{}
+	var allMissingPolicyRules []ScopedPolicyRules
 	dm, err := a.decodeManifest(manifestReader)
 	if err != nil {
 		return nil, err
 	}
 	attributesRecords := dm.asAuthorizationAttributesRecordsForUser(manifestManager)
 
-	var preAuthEvaluationErrors []error
+	// Compute missing rules using your original logic.
 	missingRules, err := a.authorizeAttributesRecords(ctx, attributesRecords)
 	if err != nil {
-		preAuthEvaluationErrors = append(preAuthEvaluationErrors, err)
+		// If there are errors here, you might choose to log them or collect them.
+		// We'll still try to run the upstream escalation check.
 	}
 
+	var escalationErrors []error
 	ec := escalationChecker{
 		authorizer:        a.authorizer,
 		ruleResolver:      a.ruleResolver,
 		extraClusterRoles: dm.clusterRoles,
 		extraRoles:        dm.roles,
 	}
-
 	for _, obj := range dm.rbacObjects() {
 		if err := ec.checkEscalation(ctx, manifestManager, obj); err != nil {
-			// In Kubernetes 1.32.2 the specialized PrivilegeEscalationError is gone.
-			// Instead, we simply collect the error.
-			preAuthEvaluationErrors = append(preAuthEvaluationErrors, err)
+			// If err is a composite error (errors.Join), unwrap it and add each underlying error.
+			var joinErr interface{ Unwrap() []error }
+			if errors.As(err, &joinErr) {
+				for _, singleErr := range joinErr.Unwrap() {
+					escalationErrors = append(escalationErrors, singleErr)
+				}
+			} else {
+				escalationErrors = append(escalationErrors, err)
+			}
 		}
 	}
 
-	for ns, nsMissingRules := range missingRules {
-		// NOTE: Although CompactRules is defined to return an error, its current implementation
-		// never produces a non-nil error. This is because all operations within the function are
-		// designed to succeed under current conditions. In the future, if more complex rule validations
-		// are introduced, this behavior may change and proper error handling will be required.
-		if compactMissingRules, err := validation.CompactRules(nsMissingRules); err == nil {
-			missingRules[ns] = compactMissingRules
+	// If escalation check fails, return the detailed missing rules along with the error.
+	if len(escalationErrors) > 0 {
```

Comment:
The `missingRules` are separate from the escalation check. If there are `missingRules`, but there are no escalation errors, we still need to report the missing rules.

For example, consider a manifest where the only contents are a secret and configmap. There will never be an escalation there because only RBAC objects can fail escalation checks. If a user lacks permissions on either secrets or configmaps, we need to fail return those and fail the preflight check.

Reply:
@bentito similarly, just double checking before closing outdated stuff